commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/MercanetCommercePaymentMethod.java

```diff
@@ -224,8 +224,8 @@ public CommercePaymentResult processPayment(
 
 		String url = StringBundler.concat(
 			_getServletUrl(mercanetCommercePaymentRequest), StringPool.QUESTION,
-			"redirectUrl=", redirectionUrl.toString(), StringPool.AMPERSAND,
-			"redirectionData=",
+			"redirectUrl=", _http.encodeURL(redirectionUrl.toString()),
```

Comment:
@luca-pellizzon MAke sure we always encode urls in other payments methods too, adn when ever we do creare the redirects urls